Cargo.toml

```diff
@@ -103,7 +103,7 @@ num-integer = "0.1.44"
 num-traits = "0.2.14"
 openblas-src = { version = "0.10", features = ["static"] }
 paste = "1.0.5"
-proptest = "1.0.0"
+proptest = { version = "1.0.0", default-features = false, features = ["std"] }
```

Comment:
what feature do we loose here ? is that gonna be a problem ?

Reply:
I pushed a new change to enable some of the default features that are compatible with Wasm.

After that change, for `criterion` we will lose only `rayon`:
https://github.com/bheisler/criterion.rs/blob/f1ea31a92ff919a455f36b13c9a45fd74559d0fe/Cargo.toml#L74C12-L74C54

For `proptest` we will lose `fork` and `timeout`:
https://github.com/proptest-rs/proptest/blob/a62a348b59f422161cbc5c6910f83f1b3c3e67e5/proptest/Cargo.toml#L22

These are multi-threading features that affect performance.

If that's unacceptable, I can revert all `Cargo.toml` changes and keep the current status quo of not running Wasm tests. I think that would be okay because we can require people to run those tests locally when making Wasm-related changes.

Reply:
We really need to keep proptest fork around (that's super useful to test kernel under address sanitizing, as the sanitizer will abort the process at the first issue). Can we achieve keeping proptest whole on every non-wasm platform by some cfg() tricks in Cargo.toml ?

Reply:
I was also thinking in that direction initially. I couldn't find a way to have a platform-specific dependency in the workspace: https://github.com/rust-lang/cargo/issues/5220

I could explore moving `proptest` into a create-local dependency (instead of workspace) and then it should be possible to have platform-specific rules.

Reply:
Worth a try. If it solves the problem, I'm ok dealing proptest becoming a crate-level dep.

Reply:
This should be fixed in the latest commit.